src/scenario.cpp

```diff
@@ -221,6 +224,74 @@ start_location_id scenario::random_start_location() const
     return random_entry( _allowed_locs );
 }
 
+bool scenario::scen_is_blacklisted() const
+{
+    return sc_blacklist.scenarios.count( id ) != 0;
+}
+
+void scen_blacklist::load_scen_blacklist( const JsonObject &jo, const std::string &src )
+{
+    sc_blacklist.load( jo, src );
+}
+
+void scen_blacklist::load( const JsonObject &jo, const std::string & )
+{
+    if( !scenarios.empty() ) {
+        jo.throw_error( "Attempted to load scenario blacklist with an existing scenario blacklist" );
+    }
+
+    std::string bl_stype;
+    if( jo.has_string( "subtype" ) ) {
```

Comment:
Don't check for this. Just read the string via `get_string`. If there is no member of that name, or if it has an incompatible type, an error will be generated. You don't need  to do that  yourself.